### DISCUSSION TITLE
Is it possible to add a feature which can generate slide's thumbnail

### DISCUSSION THREAD

**Discussion by BiDuang** [ShapeCrawler/ShapeCrawler#588] [Q&A] [answered]
I knew that `Microsoft.Office.Interop.PowerPoint` provided a method can export slide as `png` image, but is it possible to make a feature that could export slide as image without having Microsoft Office installed.

**Comment by AdamShakhabov** [comment]
No, this feature is not available yet. It is still in planning and waiting for a volunteer🙂

**Reply by BiDuang** [reply]
My current solution is use package `Microsoft.Office.Interop.PowerPoint` and call method `Export()`. But obviously it will cause this project can not work without Microsoft Office installed anymore. So I'm finding a way that could generate images for slides.

**Reply by AdamShakhabov** [reply, marked as answer]
I haven't looked for an _open-source_ C# library that can generate images for slides in a long time. But I think there isn't such a tool available yet.

The last time I needed it for a production application, I used a commercial library for that purpose.

And yes, Interop.PowerPoint should not be used on the back-end side. It's a pain🙂

**Reply by BiDuang** [reply]
I got an idea: what about to use `LibreOffice`? LibreOffice provides a way to export slide as image, and it's free and opensource.

**Reply by AdamShakhabov** [reply]
If I remember correctly, it also requires the LibreOffice application to be installed, right?

**Reply by BiDuang** [reply]
Is it possible to reverse the source code?

**Reply by AdamShakhabov** [reply]
"reverse"? You mean converting its C/C++ into C#?

**Reply by BiDuang** [reply]
Yes, so it will be a hard work. 😭  
I'll looking for an easier way.

**Reply by AdamShakhabov** [reply]
Yes, it is difficult. 😶 I've tried using the source code of LibreOffice as documentation for the slide image generation feature or, for example, when developing chart processing features. But as I mentioned, it's complicated.

**Reply by BiDuang** [reply]
Okay, maybe we could find a solution in future. I'll close this discussion as duplicate.🥲

**Comment by BiDuang** [comment]
> If I remember correctly, it also requires the LibreOffice application to be installed, right?
